views/index.ejs

```diff
@@ -7,31 +8,59 @@
     <link rel="icon" href="/public/favicon.png" />
     <link rel="stylesheet" href="/public/style.css" />
 </head>
+
 <body>
     <main>
         <div class="card">
             <% // The serverList is passed in from the clientApp %>
-            <% serverList.forEach(section => { %>
-                <div class="card-section" data-section="<%= section.name %>">
-                    <% if (section.name.toLowerCase() === 'official') { %>
-                        <div class="screeps-logo">
-                            <img src="<%= section.logo %>" alt="Screeps Logo" />
-                        </div>
-                    <% } else { %>
-                        <h2><%= section.name %></h2>
+                <% if(serverList.length===0){ %>
+                    <h1> Could not load Server List</h1>
+                    <% } %>
+                        <% serverList.forEach(section=> { %>
+                            <div class="card-section" data-section="<%= section.name %>">
+                                <% if (section.name.toLowerCase()==='official' ) { %>
+                                    <div class="screeps-logo">
+                                        <img src="<%= section.logo %>" alt="Screeps Logo" />
+                                    </div>
+                                    <% } else { %>
+                                        <h2>
+                                            <%= section.name %>
+                                        </h2>
+                                        <% } %>
+                                            <% section.servers.forEach(server=> { %>
+                                                <div class="server">
+                                                    <a href="<%= server.url %>">
+                                                        <button>
+                                                            <%= server.name %>
+                                                        </button>
+                                                    </a>
+                                                    <span class="status-circle" data-api="<%= server.api %>"></span>
+                                                </div>
+                                                <% }) %>
+                            </div>
+                            <% }) %>
+        </div>
```

Comment:
Indentation gets a little wacky here. Unfortunately the auto formatter doesn't handle these .ejs files well.

Reply:
@Gadjung

Reply:
should be fixed